### PR TITLE
[Feature]: Adding Stargazers  animated gif

### LINKED ISSUE #1557: [Feature]: Adding a stargazers generated video to showcase the popularity of the repo
### Feature Name

Adding a stargazers type like mentioned below 

### Feature Description

This is the type of change I want to introduce to make this appear here in the empty stage here

image

This video will represent a good feedback of **why choose Algo ?**

https://github.com/user-attachments/assets/0a4fc960-cd7e-40a5-b3c5-a5a2336c54e0

### Motivation

Motivation is because your repo is really good for us contributors to contribute and learn the basics of the Git & Github platforms, tools and engage with the community too.

### Implementation Suggestions (Optional)

To simple edit the main index.md to access the video file from the assets or video folder.

### Feature Type

UI/UX Improvement

### Does this feature require additional resources?

- [ ] Additional Documentation
- [ ] Third-Party Libraries
- [X] UI/UX Design
- [ ] Other

### References (Optional)

_No response_

### PR DESCRIPTION
**Pull request by yashksaini-coder**
### Description
Add a new component `AnimatedGIF.tsx` to render & display `gif` & `mp4` files easily.
Also exported the stargazers animation video.

Fixes #1557 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

---

@ajay-dhangar can you review this PR, and add proper labels to it. Also assign this issue to me.

### CODE REVIEW

**Comment by yashksaini-coder** [comment]
Here is the final result:- 

[algo-1557.webm](https://github.com/user-attachments/assets/2a25fef6-d897-4efd-b757-a6f696017ae4)

File: src/components/Homepage/HeroSection.tsx [hunk 1]
```diff
@@ -14,18 +16,22 @@ const HeroSection: React.FC = () => {
...
         </p>
         <div className="flex flex-col md:flex-row items-center justify-center space-y-4 md:space-y-0 md:space-x-6">
-          <Link
-            to="/docs"
-            className="flex items-center px-6 py-3 text-lg font-medium hover:text-white text-white bg-blue-600 hover:bg-blue-700 rounded-lg transition duration-300 ease-in-out shadow-lg transform hover:scale-105"
+          <a
+            href="/docs"
+            className="flex items-center px-6 py-3 text-lg font-medium text-white bg-blue-600 hover:bg-blue-700 rounded-lg transition duration-300 ease-in-out shadow-lg transform hover:scale-105"
           >
             Explore Algorithms <FaArrowRight className="ml-2" />
```
Comment (ajay-dhangar) [inline src/components/Homepage/HeroSection.tsx]: Replace `a` with `Link` and `href` with `to`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Our repo is `https://github.com/ajay-dhangar/algo/`. So, can you modify based on our contents and resources

**Comment by yashksaini-coder** [comment]
@ajay-dhangar I have done the requested changes to the `HeroSection.tsx`

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**